glib/src/object.rs

```diff
@@ -2094,6 +2095,20 @@ pub trait ObjectExt: ObjectType {
     /// Downgrade this object to a weak reference.
     fn downgrade(&self) -> WeakRef<Self>;
 
+    // rustdoc-stripper-ignore-next
+    /// Add a callback to be notified when the Object is disposed.
+    #[doc(alias = "g_object_weak_ref")]
+    fn add_weak_ref_notify<F: FnOnce() + Send + Sync + 'static>(&self, f: F)
```

Comment:
The feature to get a callback when the object is disposed is literally called "weak ref" in GObject: `g_object_weak_ref()`

Reply:
Sure, if that's the way GObject calls this feature we can use it. I was just looking at this from the perspective of a "normal" Rust programmer.

Also, I've seen that this PR hasn't seen a lot of activity lately, I'd be happy if I can help to get it across the finish line :)

Reply:
I don't like all the names so far, so I would stay with the GObject terminology. But if someone has a better idea. In Rust terms this would be a "drop notify" but that's also sounding strange.

Reply:
@AaronErhardt Also I think you can take this up if you want. @jsparber seems to be very busy and there's not much left here anyway.

OOC where do you plan to use this?

Reply:
> OOC where do you plan to use this?

The in-development version of Relm4 requires the root of a component to implement a helper trait called `OnDestroy` which is implemented using `WidgetExt::connect_destroy`. This mechanism makes it possible to bind the runtime of the component to the lifetime of its root widget. However, this is problematic for native widgets which don't implement `WidgetExt` because they cannot be the root of a component in this case. The plan is to implement `OnDestroy` for all objects via `ObjectExt` to remove this limitation.

Relevant issue: https://github.com/Relm4/Relm4/issues/143

Reply:
You won't get the object itself in the callback though, only a pointer to the already gone object. That's sufficient for you?

Also an alternative to this is to use `ObjectExt::set_data()` and similar and to go via the `Drop` impl of the stored data.

Reply:
Yes that should be sufficient. It's just used to send a final message to the component that tells it to shut down its runtime.

The current implementation of `OnDestroy` can be found here: https://github.com/Relm4/Relm4/blob/next/relm4/src/component/mod.rs#L48